File: LeetCode C++ Solutions/container-with-most-water.cpp
/**
 * Author: Siddhant Pandya
 * Problem: 11. Container With Most Water
 */

class Solution {
public:
    int maxArea(vector<int>& height) {
        int area=0;
        int l=0, r=height.size()-1;
        while(l<r)
        {
            int temph=min(height[l],height[r]);
            if(area<(temph*(r-l)))
            {
                area=temph*(r-l);
            }
            
            if(height[l]<height[r])
            {
                l++;
            }
            else
            {
                r--;
            }
        }
        return area;
    }
};

File: Java/ContainerWithMostWater.java
public class ContainerWithMostWater{
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length -1;
        int max = Integer.MIN_VALUE;
     
        while(left < right){
            int min = Math.min(height[left], height[right]);//contraint
            max = Math.max(max, min *(right-left));
            if(height[left] > height[right])
                right--;
            else
                left++;

           
        }
        return max;
    }

    public static void main(String[] args) {
        ContainerWithMostWater maxDimensions = new ContainerWithMostWater();
        int[] height = {1,8,6,2,5,4,8,3,7};
        System.out.println("Container that can hold the most water's size is " + maxDimensions.maxArea(height));
        
    }


}